commit 903f334ef06fbec28d23b7c8ea3c79cacc1da14f
Author: Mark Wielaard <mjw@redhat.com>
Date:   Tue May 27 10:25:27 2014 +0200

    readelf: Skip section if name is NULL in print_debug.
    
    Don't crash and burn when a section doesn't have a name (possibly invalid
    ELF file string table). Just try the next section instead of calling strcmp
    on NULL.
    
    Signed-off-by: Mark Wielaard <mjw@redhat.com>

diff --git a/src/readelf.c b/src/readelf.c
index ea14a09..6bbe436 100644
--- a/src/readelf.c
+++ b/src/readelf.c
@@ -7962,8 +7962,10 @@ print_debug (Dwfl_Module *dwflmod, Ebl *ebl, GElf_Ehdr *ehdr)
 				       / sizeof (debug_sections[0]));
 	  const char *name = elf_strptr (ebl->elf, shstrndx,
 					 shdr->sh_name);
-	  int n;
+	  if (name == NULL)
+	    continue;
 
+	  int n;
 	  for (n = 0; n < ndebug_sections; ++n)
 	    if (strcmp (name, debug_sections[n].name) == 0
 #if USE_ZLIB
